Plugins/Public/NoDock/Main.cpp

```diff
@@ -0,0 +1,574 @@
+// A map of our ID internal name and all that it can no dock | <ID, <SpaceObj>>
+map<uint, list<uint>> mapIDRestrictions; 
+
+// A map of our internal name and all that it is restricted from docking with | <ID, <SpaceObj>>
+map<uint, list<uint>> mapIDRestricted;
+
+// A map of active nodocks | <iClientID, <ID, Duration>>
+map<uint, map<uint, int>> mapActiveNoDocks;
+
+// List of ships ignored by the /nodock command (stop people nodocking admins)
+list<wstring> lstIgnoredShips;
+
+/// A return code to indicate to FLHook if we want the hook processing to continue.
+PLUGIN_RETURNCODE returncode;
+
+void LoadSettings();
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+	srand((uint)time(0));
+	// If we're being loaded from the command line while FLHook is running then
+	// set_scCfgFile will not be empty so load the settings as FLHook only
+	// calls load settings on FLHook startup and .rehash.
+	if(fdwReason == DLL_PROCESS_ATTACH)
+	{
+		if (set_scCfgFile.length()>0)
+			LoadSettings();
+	}
+	else if (fdwReason == DLL_PROCESS_DETACH)
+	{
+	}
+	return true;
+}
+
+/// Hook will call this function after calling a plugin function to see if we the
+/// processing to continue
+EXPORT PLUGIN_RETURNCODE Get_PluginReturnCode()
+{
+	return returncode;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Logging
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+FILE *Logfile = fopen("./flhook_logs/nodockcommand.log", "at");
+
+void Logging(const char *szString, ...)
+{
+	char szBufString[1024];
+	va_list marker;
+	va_start(marker, szString);
+	_vsnprintf(szBufString, sizeof(szBufString) - 1, szString, marker);
+
+	if (Logfile) {
+		char szBuf[64];
+		time_t tNow = time(0);
+		struct tm *t = localtime(&tNow);
+		strftime(szBuf, sizeof(szBuf), "%d/%m/%Y %H:%M:%S", t);
+		fprintf(Logfile, "%s %s\n", szBuf, szBufString);
+		fflush(Logfile);
+		fclose(Logfile);
+	}
+	else {
+		ConPrint(L"Failed to write nodockcommand log! This might be due to inability to create the directory - are you running as an administrator?\n");
+	}
+	Logfile = fopen("./flhook_logs/nodockcommand.log", "at");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Loading Settings
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LoadSettings()
+{
+	returncode = DEFAULT_RETURNCODE;
+	int noDockIDS = 0; // How many IDs are we going to have
+	int noDockedBases = 0; // How many bases can be nodocked
+	int restrictedCount = 0; // How many solars are restricted
+	int ignoredShips = 0; // How many ships are ignored by nodock
+
+	char szCurDir[MAX_PATH];
+	GetCurrentDirectory(sizeof(szCurDir), szCurDir);
+	string NoDockCfg = string(szCurDir) + R"(\flhook_plugins\laz_NoDock.cfg)"; // Our Config File
```

Comment:
waho what is going on here?

Reply:
The entire thing is in a new project since you mentioned you desired projects to be separated rather than merged. This seemed like a good opportunity.

Reply:
Yeah my point was the raw string thing. I hadn't seen it before, but it's okay.